### PR TITLE
New unary option `readOnly` to declare an entity as read-only

### LINKED ISSUE #318: Add JDL keyword "readonly"
Hi guys.

In reference to this previous closed post https://github.com/jhipster/generator-jhipster/issues/8800

Myself and a few others on our team are thinking about working on this as its a use case thats common in our company.

Having never worked on an open source project before, can anyone point me to in the codebase some examples of a  keyword in JDL and how it effects what is generated? functionality we can use as a base? At its core its simply adding @ Immutable to the JPA class and not generating certain stuff which seems simple enough

### PR DESCRIPTION
**Pull request by murdos**
Fixes #318

I used the keywork `readOnly`, but only at end I realized that it could have been `readonly` (both are used in #318).
Please let me know what you prefer, and I'll update the PR.

Please make sure the below checklist is followed for Pull Requests.
  - [x] [Travis tests](https://travis-ci.org/jhipster/jhipster-core/pull_requests) are green
  - [X] Tests are added where necessary
  - [X] Documentation is added/updated where necessary
  - [X] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/jhipster-core/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by MathieuAA** [comment]
Thanks @murdos I'll test this out ASAP!

File: lib/converters/json_to_jdl_entity_converter.js [hunk 1]
```diff
@@ -316,6 +316,9 @@ function addEntityOptionsToJDL(entity, entityName) {
   if (entity.jpaMetamodelFiltering === true) {
     addUnaryOptionToJDL(FILTER, entityName);
   }
+  if (entity.readOnly === true) {
```
Comment (MathieuAA) [inline lib/converters/json_to_jdl_entity_converter.js]: missing a test here:
  - add in the Employee.json file (used in this converter's test file) this entry "readOnly": true.
  - in the test, l.156, add this:
```javascript
          expect(
            jdlObject
              .getOptions()
              .filter(option => option.name === UnaryOptions.READ_ONLY && option.entityNames.has('Employee')).length
          ).to.equal(1);
```

**Comment by murdos** [comment]
Thanks. I'm going to add the missing test.
So do we keep with `readOnly` or `readonly`?

**Comment by MathieuAA** [comment]
readOnly is good, let's keep it this way

**Comment by murdos** [comment]
Ok. Missing test has been added

**Comment by MathieuAA** [comment]
Thanks :)

**Comment by MathieuAA** [comment]
Alright, let's merge it. Thanks @murdos!

**Pull request merged**